Client/mods/deathmatch/logic/luadefs/CLuaVectorGraphicDefs.cpp

```diff
@@ -0,0 +1,318 @@
+                                                       std::optional<CLuaFunctionRef> luaFunctionRef)
+{
+    if (size.fX <= 0 || size.fY <= 0)
+        throw std::invalid_argument("A vector graphic must be atleast 1x1 in size.");
+
+    if (size.fX > 4096 || size.fY > 4096)
+        throw std::invalid_argument("A vector graphic cannot exceed 4096x4096 in size.");
+
+    CLuaMain&  pLuaMain = lua_getownercluamain(luaVM);
+    CResource* pParentResource = pLuaMain.GetResource();
+
+    CClientVectorGraphic* pVectorGraphic =
+        g_pClientGame->GetManager()->GetRenderElementManager()->CreateVectorGraphic(static_cast<int>(size.fX), static_cast<int>(size.fY));
+
+    if (!pVectorGraphic)
+        return false;
+
+    if (pathOrRawData.has_value())
+    {
+        std::string& strRawData = pathOrRawData.value();
+        std::size_t  szCharPos = strRawData.find_first_not_of(" \t\n\r\f\v", 0);
+
+        char charAuto = 0;
+
+        if (szCharPos != std::string::npos)
+            charAuto = strRawData.at(szCharPos);
+
+        if (charAuto && charAuto == '<')
+        {
+            if (luaFunctionRef.has_value() && VERIFY_FUNCTION(luaFunctionRef.value()))
+            {
+                CLuaFunctionRef funcRef = luaFunctionRef.value();
+
+                CLuaShared::GetAsyncTaskScheduler()->PushTask<bool>(
+                    [funcRef, pVectorGraphic, strRawData] {
+                        lua_State* luaVM = funcRef.GetLuaVM();
+                        CLuaMain*  pLuaMain = m_pLuaManager->GetVirtualMachine(luaVM);
+
+                        if (!pLuaMain || !pVectorGraphic)
+                            return false;
+
+                        if (!pVectorGraphic->LoadFromString(strRawData))
+                        {
+                            delete pVectorGraphic;
+
+                            m_pScriptDebugging->LogCustom(luaVM, "Unable to load SVG data (check for XML syntax errors)");
+                            return false;
+                        }
+
+                        return true;
+                    },
+                    [funcRef](const bool didLoad) {
+                        CLuaMain* pLuaMain = m_pLuaManager->GetVirtualMachine(funcRef.GetLuaVM());
+                        if (pLuaMain)
+                        {
+                            CLuaArguments arguments;
+                            arguments.PushBoolean(didLoad);
+                            arguments.Call(pLuaMain, funcRef);
+                        }
+                    });
+            }
+            else
+            {
+                if (!pVectorGraphic->LoadFromString(pathOrRawData.value()))
+                {
+                    delete pVectorGraphic;
+
+                    m_pScriptDebugging->LogCustom(luaVM, "Unable to load SVG data (check for XML syntax errors)");
+                    return false;
+                }
+            }
```

Comment:
The code is currently duplicated twice, once for the callback and once for the non-callback. I would like this to be refactored into a e.g. single private function call.